### DISCUSSION TITLE
Recursive Query with cycles

### DISCUSSION THREAD

**Discussion by RaulPampliegaMayoral** [Blazebit/blaze-persistence#1798] [Q&A] [answered]
Hi all!

We are using a recursive query correctly but in certain occasions the structures can be generated with cycles (the client is the one that generates this data). Is there any way to avoid these cycles with Blaze Persistence?

![image](https://github.com/Blazebit/blaze-persistence/assets/13700946/a2b5ccb0-768b-4948-86b4-3bc0be09ff91)

**Comment by beikov** [comment, marked as answer]
Hi there. One way to avoid cycles is to check in the recursive part of your query whether the path already contains the current element i.e. `where path not like concat('%', label, '/%')`. I think it should also be possible to use `union` instead of `union all` to avoid duplicates.

**Comment by RaulPampliegaMayoral** [comment]
Hi @beikov!
Thank you for your response! The solution works perfectly, I should have noticed it myself!
Thanks again for your help!
